prune.go

```diff
@@ -225,12 +259,12 @@ func PruneTxns(db *mgo.Database, txns *mgo.Collection) error {
 			logger.Debugf("%d completed txns pruned so far", remover.removed)
 		}
 	}
-	if err := remover.flush(); err != nil {
-		return fmt.Errorf("removing txns: %v", err)
-	}
 	if err := iter.Close(); err != nil {
 		return fmt.Errorf("iterating through unreferenced txns: %v", err)
 	}
+	if err := remover.flush(); err != nil {
```

Comment:
moved back. mostly I did a pass because I realized I wasn't closing my own iterators, but it makes sense.
I'm a little worried the error may be "db in unknown state", but I agree that a NotFound should let us flush the removals before returning.